routes/api-routes.js

```diff
@@ -2,9 +2,19 @@ const router = require('express').Router();
 
 // import controllers here...
 const postsController = require('../controllers/postsController.js');
+const usersController = require('../controllers/usersController.js');
 
 // routes and route handlers (controllers)
+
+//for the posts
 router.get('/getPosts', postsController.getPosts);       // get all posts
 router.get('/viewPost/:pid', postsController.viewPost);  // get single post
-// export
+router.get('/getGallery', postsController.getGallery);
+
+//for the users
+router.get('/profile' , usersController.getProfile);
```

Comment:
this should be done on the routes file not in the API.

API is only for retrieving data. Kaya kung mapapansin mo, lahat yan res.send(). 

Think of it this way, if you need it to be in AJAX request, use API.
If it should be rendered by the server before sending to client, you don't need API. The controller is correct but the route handler must not be in API file. I'm pertaining to this code: `router.get('/profile' , usersController.getProfile);`